core/src/main/java/com/linecorp/armeria/server/annotation/AnnotatedService.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.annotation;
+
+import java.lang.reflect.Method;
+
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.Route;
+import com.linecorp.armeria.server.ServerBuilder;
+
+/**
+ * An {@link HttpService} which is defined by a {@link Path} or HTTP method annotations.
+ * This class is designed to provide a common interface for {@link AnnotatedService}
+ * internally. Please check out the documentation at
+ * <a href="https://armeria.dev/docs/server-annotated-service">Annotated HTTP Service</a> to use this.
+ */
+public interface AnnotatedService extends HttpService {
```

Comment:
Let's add `@UnstableApi` to new public APIs.
```suggestion
@Nullable
public interface AnnotatedService extends HttpService {
```